### PR TITLE
Am/feat/zk metadata

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Comment by IceTDrinker** [comment]
Check for the wasm test change https://github.com/zama-ai/tfhe-rs/actions/runs/10806528752

**Comment by IceTDrinker** [comment]
new run https://github.com/zama-ai/tfhe-rs/actions/runs/10807698866 (forgot to format something in the previous one)

**Comment by IceTDrinker** [comment]
test is failing locally :/ we don't get the feedback, we need to get this python test runne ASAP

File: tfhe/src/core_crypto/algorithms/test/lwe_encryption.rs [hunk 1]
```diff
@@ -1012,6 +1012,8 @@ fn lwe_compact_public_encrypt_prove_verify_decrypt_custom_mod<Scalar>(
     let message_modulus_log = params.message_modulus_log;
     let encoding_with_padding = get_encoding_with_padding(ciphertext_modulus);
 
+    let metadata = [b'c', b'o', b'r', b'e'];
```
Comment (nsarlin-zama) [inline tfhe/src/core_crypto/algorithms/test/lwe_encryption.rs:1015]: could also be expressed as `"core".as_bytes()`

File: tfhe/c_api_tests/test_high_level_zk.c [hunk 2]
```diff
@@ -35,6 +35,11 @@ int main(void) {
   status = compact_pke_crs_public_params(crs, &public_params);
   assert(status == 0);
 
+  // This is a hack to have metadata_len be "constant" i.e. cannot change between compilation and
+  // runtime, otherwise metadata[metadata_len] is considered a Variable Length Array
+  enum { metadata_len = (size_t)5 };
```
Comment (nsarlin-zama) [inline tfhe/c_api_tests/test_high_level_zk.c:40]: how about `#define METADATA_LEN 5` ?

**Review by nsarlin-zama — COMMENTED** [review]
some small style comments, but it looks good to me

File: tfhe/src/core_crypto/algorithms/test/lwe_encryption.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/test/lwe_encryption.rs:1015]: hmm, I don't know if I want to go with that style given utf-8 is variable length

File: tfhe/c_api_tests/test_high_level_zk.c (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/c_api_tests/test_high_level_zk.c:40]: \*sigh\*

why can't we have nice things, macros for array len 😭

**Review by nsarlin-zama — APPROVED** [review]

**Comment by IceTDrinker** [comment]
https://github.com/zama-ai/tfhe-rs/actions/runs/10814769158

**Pull request merged**